tests/common/reboot.py

```diff
@@ -143,34 +150,35 @@ def execute_reboot_helper():
     logger.info('ssh has started up')
 
     logger.info('waiting for switch to initialize')
-    time.sleep(wait)
-    DUT_ACTIVE.set()
 
     if reboot_type == 'warm':
+        logger.info('waiting for warmboot-finalizer service to become activating')
+        finalizer_state = get_warmboot_finalizer_state(duthost)
+        while finalizer_state != 'activating':
```

Comment:
Are we guaranteed that we will be in activating state and not reached inactive state at this point?

Reply:
There should be a guarantee, as I have removed the `time.sleep(wait)` from the existing code.
The modified code checks for `activating` state as soon as SSH comes up. I think we cannot check this any sooner than this.

Reply:
Lets go ahead with this change and revisit if we encounter any issues